tests/test_interface.py

```diff
@@ -37,6 +37,23 @@ def list_pending_parts(self, jobid, metdata):
         return []
 
 
+@pytest.fixture
+def CustomCallback():
+    class OnReduce():
+        def __init__(self):
+            self.called = False
+            pass
+        def on_reduce(self, message, topic, subject):
+            self.called = True
+            self.message = message
+            self.topic = topic
+            self.subject = subject
+        def assert_called_once(self):
+            assert self.called
+
+    return OnReduce()
```

Comment:
I've never seen this pattern for mocking a function. It totally works but might be less code to use `mock`? 

```
from mock import Mock
reduce_callback = Mock()

Part(..., on_reduce=reduce_callback):
    pass

reduce_callback.assert_called_once_with(message, topic, subject)
```

We can adjust later, let's not hold up this PR.

Reply:
I'll give this a try

Reply:
Works great!